packages/devtools_app/lib/src/service/resolved_uri_manager.dart

```diff
@@ -0,0 +1,51 @@
+import '../shared/globals.dart';
+import 'vm_service_wrapper.dart';
+
+/// Manager for handling package Uri lookup and caching.
+class ResolvedUriManager {
+  Map<String, String?>? _resolvedUrlMap;
+  VmServiceWrapper? service;
+
+  /// Initializes the [ResolvedUriManager]
+  void vmServiceOpened() {
+    _resolvedUrlMap = <String, String?>{};
+  }
+
+  /// Cleans up the resources of the [ResolvedUriManager]
+  void vmServiceClosed() {
+    _resolvedUrlMap = null;
```

Comment:
Style nit: this can be a one-liner (not sure if we have preference one way or another in DevTools, @kenzieschmoll may have an opinion).